pam/pam-module-transaction-dummy.go

```diff
@@ -0,0 +1,295 @@
+package main
+
+import "C"
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+	"unsafe"
+
+	"github.com/msteinert/pam"
+)
+
+// PamModuleTransactionDummy is an implementation of PamModuleTransaction for
+// testing purposes.
+type PamModuleTransactionDummy struct {
+	Items       map[pam.Item]string
+	Env         map[string]string
+	Data        map[string]any
+	convHandler pam.ConversationHandler
+}
+
+// NewPamModuleTransactionDummy returns a new PamModuleTransactionDummy.
+func NewPamModuleTransactionDummy(convHandler pam.ConversationHandler) pam.ModuleTransaction {
+	return &PamModuleTransactionDummy{
+		convHandler: convHandler,
+	}
+}
+
+// InvokeHandler is called by the C code to invoke the proper handler.
+func (m *PamModuleTransactionDummy) InvokeHandler(handler pam.ModuleHandlerFunc,
+	flags pam.Flags, args []string) (pam.Status, error) {
+	return pam.Abort, pam.Abort
+}
+
+// SetItem sets a PAM information item.
+func (m *PamModuleTransactionDummy) SetItem(item pam.Item, value string) error {
+	if m.Items == nil {
+		m.Items = make(map[pam.Item]string)
+	}
+
+	m.Items[item] = value
+	return nil
+}
+
+// GetItem retrieves a PAM information item.
+func (m *PamModuleTransactionDummy) GetItem(item pam.Item) (string, error) {
+	if m.Items == nil {
+		return "", nil
+	}
+	return m.Items[item], nil
+}
+
+// PutEnv adds or changes the value of PAM environment variables.
+//
+// NAME=value will set a variable to a value.
+// NAME= will set a variable to an empty value.
+// NAME (without an "=") will delete a variable.
+func (m *PamModuleTransactionDummy) PutEnv(nameVal string) error {
+	if m.Env == nil {
+		m.Env = make(map[string]string)
+	}
+
+	env, value, found := strings.Cut(nameVal, "=")
+	if !found {
+		delete(m.Env, env)
+		return nil
+	}
+	if env == "" {
+		return pam.BadItem
+	}
+	m.Env[env] = value
+	return nil
+}
+
+// GetEnv is used to retrieve a PAM environment variable.
+func (m *PamModuleTransactionDummy) GetEnv(name string) string {
+	if m.Env == nil {
+		return ""
+	}
+
+	return m.Env[name]
+}
+
+// GetEnvList returns a copy of the PAM environment as a map.
+func (m *PamModuleTransactionDummy) GetEnvList() (map[string]string, error) {
+	if m.Env == nil {
+		m.Env = make(map[string]string)
+	}
+	return m.Env, nil
+}
```

Comment:
IMHO, we shouldn't assign nor create anything inside a "getter" function, only return the requested value and leave it up to the caller to check if it isn't empty and/or nil. (i.e. only `return m.Env, nil`)

Reply:
Oh, this got hidden... Addressed now.